hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/HandleWorkflow.java

```diff
@@ -161,6 +167,16 @@ private void handlePlatformTransaction(
                     .transactionID(txBody.transactionID())
                     .memo(txBody.memo());
 
+            // If pre-handle was successful, we return the result. Otherwise, we charge the node or throw an exception.
+            switch (preHandleResult.status()) {
```

Comment:
On second thought, this change is wrong. 

If we have a due diligence error or an unknown failure, `preHandleResult.txInfo()` is `null`. I don't know what the correct behavior is. What do we write into the record stream if we cannot parse the transaction?

If I remember correctly, we did not want to write anything in the record stream in the case of an unknown failure but just update the receipt.

We can keep this change for now but must revisit it once the diagram of the handle workflow with error conditions is ready.

Reply:
I agree, this entire method needs to be reworked. We have the pieces, but not in the right order.